src/main/scala/com/evolution/scache/LoadingCache.scala

```diff
@@ -508,9 +635,24 @@ private[scache] object LoadingCache {
     }
   }
 
+  sealed trait EntryState[F[_], A] {
+    def getOption(implicit F: Applicative[F]): F[Option[Entry[F, A]]] =
+      this match {
+        case EntryState.Loading(deferred) => deferred.getOption
+        case EntryState.Value(entry)      => entry.some.pure[F]
+        case EntryState.Removed(_)        => none[Entry[F, A]].pure[F]
+      }
+  }
+
+  object EntryState {
+    final case class Loading[F[_], A](deferred: Deferred[F, Either[Throwable, Entry[F, A]]]) extends EntryState[F, A]
+    final case class Value[F[_], A](entry: Entry[F, A]) extends EntryState[F, A]
+    final case class Removed[F[_], A](stub: Boolean = true) extends EntryState[F, A]
```

Comment:
minor: `sealed abstract case class` also does the trick

Reply:
`sealed abstract case class` still requires to have parameters though :(

It could also be `final case object Removed extends EntryState[Nothing, Nothing]`, but that means the type params need to be covariant (`sealed trait EntryState[+F[_], +A]`, and that doesn't work nicely for `F` and its implicits in different methods.

Reply:
it cannot be `object` because `compareAndSwap` does not work ;)

Reply:
or actually… object is ok in this case.